frontend/src/util/apiUtil.js

```diff
@@ -0,0 +1,51 @@
+function setHeaders(headers) {
+  
+  if (localStorage.jwtToken) {
+    return {
+      ...headers,
+      Authorization: localStorage.jwtToken,
+    };
+  } else {
+    return headers;
+  }
+}
+
+function genericFetchRequset(type, endpoint, data = {}, onSuccess, onError) {
+  debugger
+    let params = {
+      method: type,
+      headers: setHeaders({
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      }),
+      body: data,
+    };
+
+    if (type === "GET") {
+      debugger
+      delete params.body
+    } else if ((type === "POST" || type === "PUT") && (data.isHidden || data.email)) {
+      debugger
+      params.body = JSON.stringify(data)
+    } else {
+      debugger
+      params.headers = setHeaders({});
+    }
+
+    return fetch(endpoint, params)
+      .then((res) => {
+        ;
+        res.json().then((data) => onSuccess(data));
+      })
+      .catch((err) => {
```

Comment:
`.catch((err) => onError(err)` 
do it everywhere - if it's a single call no need for code blocks (`{}`)

Reply:
No Problem it was for debugging purposes